diff --git a/open_social/themes/socialbase/assets/js/like_and_dislike_service.min.js b/open_social/themes/socialbase/assets/js/like_and_dislike_service.min.js
index d7eed1428..72fd3e5fc 100644
--- a/open_social/themes/socialbase/assets/js/like_and_dislike_service.min.js
+++ b/open_social/themes/socialbase/assets/js/like_and_dislike_service.min.js
@@ -1 +1 @@
-!function(l,a){window.likeAndDislikeService=function(){function e(){}return e.vote=function(t,n,e){l.ajax({type:"GET",url:drupalSettings.path.baseUrl+"like_and_dislike/"+n+"/"+e+"/"+t,success:function(e){if(l("#like-container-"+n+"-"+t+" a").get(0).className=e.operation.like,l("#dislike-container-"+n+"-"+t+" a").get(0).className=e.operation.dislike,"1"==e.likes)var i=a.t(" like");else i=a.t(" likes");l("#like-container-"+n+"-"+t).nextAll(".vote__count").find("a").html(e.likes+i).attr("data-dialog-options",'{"title":"'+e.likes+i+'", "width":"auto"}')}})},e}()}(jQuery,Drupal);
\ No newline at end of file
+!function(o,i){window.likeAndDislikeService=function(){function e(){}return e.vote=function(n,l,e){o.ajax({type:"POST",url:drupalSettings.path.baseUrl+"like_and_dislike/"+l+"/"+e+"/"+n,success:function(a){if(o("#like-container-"+l+"-"+n+" a").get(0).className=a.operation.like,o("#dislike-container-"+l+"-"+n+" a").get(0).className=a.operation.dislike,"1"==a.likes)var s=i.t(" like");else s=i.t(" likes");["like","dislike"].forEach(function(e){var i="#"+e+"-container-"+l+"-"+n,t=o(i+" a");0!=t.length&&(a.operation[e]?t.addClass("voted"):t.removeClass("voted"),o(i+" .count").text(a[e+"s"]),"like"==e&&o(i).nextAll(".vote__count").find("a").html(a.likes+s).attr("data-dialog-options",'{"title":"'+a.likes+s+'", "width":"auto"}'))}),o(".region.region-highlighted").html("<div class='messages__wrapper layout-container'><div class='messages messages--"+a.message_type+" role='contentinfo'>"+a.message+"</div></div>")}})},e}()}(jQuery,Drupal);
\ No newline at end of file
diff --git a/open_social/themes/socialbase/components/03-molecules/like/like_and_dislike_service.js b/open_social/themes/socialbase/components/03-molecules/like/like_and_dislike_service.js
index 042a40eef..9634ba2f0 100644
--- a/open_social/themes/socialbase/components/03-molecules/like/like_and_dislike_service.js
+++ b/open_social/themes/socialbase/components/03-molecules/like/like_and_dislike_service.js
@@ -8,7 +8,7 @@
     function likeAndDislikeService() {}
     likeAndDislikeService.vote = function(entity_id, entity_type, tag) {
       $.ajax({
-        type: "GET",
+        type: "POST",
         url: drupalSettings.path.baseUrl + 'like_and_dislike/' + entity_type + '/' + tag + '/' + entity_id,
         success: function(response) {
           // Expected response is a json object where likes is the new number
@@ -25,7 +25,26 @@
           } else {
             var countText = Drupal.t(' likes');
           }
-          $('#like-container-' + entity_type + '-' + entity_id).nextAll('.vote__count').find('a').html(response.likes + countText).attr('data-dialog-options', '{"title":"' + response.likes + countText + '", "width":"auto"}');
+
+          ['like', 'dislike'].forEach(function (iconType) {
+            var selector = '#' + iconType + '-container-' + entity_type + '-' + entity_id;
+            var $aTag = $(selector + ' a');
+            if ($aTag.length == 0) {
+              return;
+            }
+            response.operation[iconType] ? $aTag.addClass('voted') : $aTag.removeClass('voted');
+            $(selector + ' .count').text(response[iconType + 's']);
+
+            if (iconType == 'like') {
+              $(selector).nextAll('.vote__count').find('a').html(response.likes + countText).attr('data-dialog-options', '{"title":"' + response.likes + countText + '", "width":"auto"}');
+            }
+          });
+
+          // Display a message whether the vote was registered or an error
+          // happened.
+          // @todo - this will work only for case when theme has messages in
+          // highlighted region.
+          $('.region.region-highlighted').html("<div class='messages__wrapper layout-container'><div class='messages messages--" + response.message_type + " role='contentinfo'>" + response.message + "</div></div>");
         }
       });
     };
diff --git a/open_social/themes/socialbase/templates/like/like-and-dislike-icons.html.twig b/open_social/themes/socialbase/templates/like/like-and-dislike-icons.html.twig
index 514642483..10f98e660 100644
--- a/open_social/themes/socialbase/templates/like/like-and-dislike-icons.html.twig
+++ b/open_social/themes/socialbase/templates/like/like-and-dislike-icons.html.twig
@@ -18,24 +18,29 @@
 
 <div class="vote-widget vote-widget--like-and-dislike">
   <div class="vote__wrapper">
-    <div class="vote-like type-{{ entity_type }}" id="like-container-{{ entity_type }}-{{ entity_id }}">
-      <a {{ like_attributes }} >
-        <svg class="icon-vote">
-          <use xlink:href="#icon-like"></use>
-        </svg>
-      </a>
-    </div>
-
-    <div class="vote-dislike type-{{ entity_type }}" id="dislike-container-{{ entity_type }}-{{ entity_id }}">
-      <a {{ dislike_attributes }}>Dislike</a>
-      <span class="count">{{ dislikes }}</span>
-    </div>
+    {% for type, icon in icons %}
+      <div class="vote-{{ type }} type-{{ entity_type }}"
+           id="{{ type }}-container-{{ entity_type }}-{{ entity_id }}">
+        {% if type == 'like' %}
+          <a {{ icon.attributes }} >
+            <svg class="icon-vote">
+              <use xlink:href="#icon-like"></use>
+            </svg>
+          </a>
+          <span class="count">{{ icon.count }}</span>
+        {% else %}
+          <a {{ icon.attributes }}>{{ icon.label }}</a>
+          <span class="count">{{ icon.count }}</span>
+        {% endif %}
+      </div>
+    {% endfor %}
 
     <div class="vote__count">
       {% if logged_in %}
-      <a class="use-ajax" data-dialog-options='{"title":"{{ modal_title }}","width":"auto"}' data-dialog-type="modal" href="/wholiked/{{ entity_type }}/{{ entity_id }}">
-        {% trans %}{{ likes }} like {% plural likes %} {{ likes }} likes{% endtrans %}
-      </a>
+        <a class="use-ajax" data-dialog-options='{"title":"{{ modal_title }}","width":"auto"}'
+           data-dialog-type="modal" href="/wholiked/{{ entity_type }}/{{ entity_id }}">
+          {% trans %}{{ likes }} like {% plural likes %} {{ likes }} likes{% endtrans %}
+        </a>
       {% else %}
         {% trans %}{{ likes }} like {% plural likes %} {{ likes }} likes{% endtrans %}
       {% endif %}
